test/k8s-integration/main.go

```diff
@@ -70,45 +71,40 @@ func init() {
 func main() {
 	flag.Parse()
 
-	if len(*stagingImage) == 0 && !*inProw {
-		klog.Fatalf("staging-image is a required flag, please specify the name of image to stage to")
+	if !*inProw {
+		ensureVariable(stagingImage, true, "staging-image is a required flag, please specify the name of image to stage to")
 	}
 
-	if len(*saFile) == 0 {
-		klog.Fatalf("service-account-file is a required flag")
-	}
-
-	if len(*deployOverlayName) == 0 {
-		klog.Fatalf("deploy-overlay-name is a required flag")
-	}
-
-	if len(*storageClassFile) == 0 && !*migrationTest {
-		klog.Fatalf("One of storageclass-file and migration-test must be set")
-	}
-
-	if len(*storageClassFile) != 0 && *migrationTest {
-		klog.Fatalf("storage-class-file and migration-test cannot both be set")
-	}
+	ensureVariable(saFile, true, "service-account-file is a required flag")
+	ensureVariable(deployOverlayName, true, "deploy-overlay-name is a required flag")
+	ensureVariable(testFocus, true, "test-focus is a required flag")
+	ensureVariable(gceZone, true, "gce-zone is a required flag")
 
-	if !*bringupCluster && len(*kubeFeatureGates) > 0 {
-		klog.Fatalf("kube-feature-gates set but not bringing up new cluster")
+	if *migrationTest {
+		ensureVariable(storageClassFile, false, "storage-class-file and migration-test cannot both be set")
```

Comment:
will adding a helper functions for `XOR` of variable, `OR` of variable,  `AND` of variables simplify this?

like migration and storage class could be
`ensureVariablesXOR(storageClassFile, migrationTest)` <- would be nice if it can somehow reflect the flag name for a standard error message for `XOR` type variables

Reply:
Hmm, I'm not convinced that this would make things easier to understand. It will be cleaner as in the flag handling part will have less lines of code, but I feel like it will be hard to understand whats going on when you have to think about boolean operators.

Reply:
talked offline, lets think about this some more and maybe do it in a follow-up PR if time permits